06_HelloGraphicsQueue/main.cpp

```diff
@@ -164,82 +164,49 @@ class HelloGraphicsQueueApp final : public examples::MonoDeviceApplication, publ
 
 		// In order to use images, we need to change their layout from asset::IImage::EL_UNDEFINED.
 		// Here it is done with use of barriers.
-		IGPUCommandBuffer::SImageMemoryBarrier bigImgLayoutTransitionBarrier = imgLayoutTransitionBarrier;
-		bigImgLayoutTransitionBarrier.image = bigImg;
-
-		IGPUCommandBuffer::SImageMemoryBarrier smallImgLayoutTransitionBarrier = imgLayoutTransitionBarrier;
-		smallImgLayoutTransitionBarrier.image = smallImg;
+		std::array<IGPUCommandBuffer::SImageMemoryBarrier, 2u> imgLayoutTransitionBarriers = {
+			imgLayoutTransitionBarrier,
+			imgLayoutTransitionBarrier
+		};
+		imgLayoutTransitionBarriers[0].image = bigImg;
+		imgLayoutTransitionBarriers[1].image = smallImg;
 
 		// Transit layouts of both images.
 		cmdbuf->pipelineBarrier(
 			E_PIPELINE_STAGE_FLAGS::EPSF_HOST_BIT | E_PIPELINE_STAGE_FLAGS::EPSF_ALL_COMMANDS_BIT,
 			E_PIPELINE_STAGE_FLAGS::EPSF_TRANSFER_BIT,
-			E_DEPENDENCY_FLAGS::EDF_NONE, 0u, nullptr, 0u, nullptr, 1u,
-			&bigImgLayoutTransitionBarrier
-		);
-
-		cmdbuf->pipelineBarrier(
-			E_PIPELINE_STAGE_FLAGS::EPSF_HOST_BIT | E_PIPELINE_STAGE_FLAGS::EPSF_ALL_COMMANDS_BIT,
-			E_PIPELINE_STAGE_FLAGS::EPSF_TRANSFER_BIT,
-			E_DEPENDENCY_FLAGS::EDF_NONE, 0u, nullptr, 0u, nullptr, 1u,
-			&smallImgLayoutTransitionBarrier
+			E_DEPENDENCY_FLAGS::EDF_NONE, 0u, nullptr, 0u, nullptr, 2u,
+			imgLayoutTransitionBarriers.data()
```

Comment:
its good practice to write `.size()` instead of a hardcoded literal, esp if the container knows its size anyway

Reply:
apply this to allthe sites where you do this